plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTables.java

```diff
@@ -103,4 +103,142 @@ public void testPropertiesTable()
+            assertUpdate("UPDATE " + tableName + " SET _bigint = 50 WHERE _bigint =  BIGINT '5'", 1);
+            assertUpdate("DELETE FROM " + tableName + " WHERE _date =  DATE '2019-09-08'", 1);
+            assertQuerySucceeds("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE");
+            assertQuery("SELECT count(*) FROM " + tableName, "VALUES 10");
+            assertQuery("SELECT count(*) FROM \"" + tableName + "$partitions\"", "VALUES 8");
+            assertQuery("SELECT count(partition._varchar) FROM \"" + tableName + "$partitions\"", "VALUES 6");
+            assertQuery("SELECT count(distinct partition._date) FROM \"" + tableName + "$partitions\"", "VALUES 2");
+
+            assertQuery("SHOW COLUMNS FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                            ('partition', 'row(_date date, _varchar varchar)', '', ''),
+                            ('file_count', 'bigint', '', ''),
+                            ('total_size', 'bigint', '', '')
+                            """);
+
+            assertQuery(
+                    "SELECT partition._date, partition._varchar FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                                (DATE '2019-09-09', 'b'),
+                                (DATE '2019-09-09', 'c'),
+                                (DATE '2019-09-09', 'd'),
+                                (DATE '2019-09-10', 'e'),
+                                (DATE '2019-09-10', 'f'),
+                                (DATE '2019-09-10', null),
+                                (null, 'g'),
+                                (null, null)""");
+        }
+        finally{
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
+
+    @Test
+    public void testPartitionsTableDifferentOrderFromDefinitionMultipleColumns()
+    {
+        String tableName = "test_partitions_table_different_order_from_definition_multiple_columns";
+        try {
+            assertUpdate("CREATE TABLE " + tableName + " (_bigint BIGINT, _date DATE, _varchar VARCHAR) WITH (partitioned_by = ARRAY['_varchar', '_date'] )");
+            assertUpdate("INSERT INTO " + tableName + " VALUES (0, CAST('2019-09-08' AS DATE), 'a'), (1, CAST('2019-09-09' AS DATE), 'b'), (2, CAST('2019-09-09' AS DATE), 'c')", 3);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (3, CAST('2019-09-09' AS DATE), 'd'), (4, CAST('2019-09-10' AS DATE), 'e'), (5, CAST('2019-09-10' AS DATE), 'f'), (4, CAST('2019-09-10' AS DATE), 'f')", 4);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (6, null, 'g'), (6, CAST('2019-09-10' AS DATE), null), (7, null, null), (8, null, 'g')", 4);
+            assertUpdate("UPDATE " + tableName + " SET _bigint = 50 WHERE _bigint =  BIGINT '5'", 1);
+            assertUpdate("DELETE FROM " + tableName + " WHERE _date =  DATE '2019-09-08'", 1);
```

Comment:
Believe there may be an issue with `TransactionLogParser`:

```
Caused by: java.time.format.DateTimeParseException: Text 'f' could not be parsed at index 0
	at java.base/java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:2108)
	at java.base/java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:2010)
	at java.base/java.time.LocalDate.parse(LocalDate.java:435)
	at java.base/java.time.LocalDate.parse(LocalDate.java:420)
	at io.trino.plugin.deltalake.transactionlog.TransactionLogParser.deserializeColumnValue(TransactionLogParser.java:211)
```

Reply:
You uncovered a problem in the previous `UPDATE` probably.
The `partitionValues` are scrambled.

```
{
  "add": {
    "path": "_varchar=f/_date=2019-09-10/20240325_095714_00005_6fhxu_944e1975-1cf7-45cf-82a7-b406140dd506",
    "partitionValues": {
      "_varchar": "2019-09-10",
      "_date": "f"
    },
    "size": 229,
    "modificationTime": 1711360634289,
    "dataChange": true,
    "stats": "{\"numRecords\":1,\"minValues\":{\"_bigint\":4},\"maxValues\":{\"_bigint\":4},\"nullCount\":{\"_bigint\":0}}",
    "tags": {}
  }
}
```

Reply:
Creating a PR

Reply:
The bugfix is about to land.
You can proceed to add the missing columns to the metadata table. 🙏

Reply:
btw, PR landed - feel free to rebase